src/dmifn.rs

```diff
@@ -589,15 +589,14 @@ pub fn dmi_processor_id(data: &SMBiosProcessorInformation<'_>) {
                     2 => {
                         println!(
                             "\tSignature: Family {}, Model {}, Stepping {}",
-                            (eax >> 8)
-                                & 0xF
+                            ((eax >> 8) & 0xF)
```

Comment:
Good catch, yes this fixes the bug.  The original C++ code had parenthesis here:
```cpp
		case 2: /* AMD, publication #25481 revision 2.28 */
			pr_attr("Signature", "Family %u, Model %u, Stepping %u",
				((eax >> 8) & 0xF) + (((eax >> 8) & 0xF) == 0xF ? (eax >> 20) & 0xFF : 0),
				((eax >> 4) & 0xF) | (((eax >> 8) & 0xF) == 0xF ? (eax >> 12) & 0xF0 : 0),
				eax & 0xF);
			break;
```